### PR TITLE
review requested - locking of hosts feature - do not merge

### PR DESCRIPTION
**Pull request by unorthodoxgeek**
please review the path I'm going down, suggestions are welcome. this is a working branch, I'm opening a PR in order to get more insights :smile: 

todos:
this needs to have a UI presence - how would the UI look like? where would I put the lock?
what functionality do we want to prevent if a host is locked? how would we show that?

### CODE REVIEW

File: app/models/host/base.rb [hunk 1]
```diff
@@ -321,6 +323,26 @@ def import_missing_ids
     # Provide _id aliases for consistency with the _name methods
     alias_attribute :hardware_model_id, :model_id
 
+    def locked?
```
Comment (ohadlevy) [inline app/models/host/base.rb:326]: if you would create expired at (e.g. this host valid only for one week) i think it would look nearly identical ?

Comment (unorthodoxgeek) [inline app/models/host/base.rb:326]: not sure I understand what you're asking. this is basically how this is implemented, the locked_until method returns a timestamp which represents the time the host gets unlocked (or -1 if the host is locked without a time limit)

Comment (unorthodoxgeek) [inline app/models/host/base.rb:326]: but for most use cases, I don't care when the host is getting unlocked. so it's a boolean method.

**Comment by dLobatog** [comment]
I'm not overly convinced about the feature in core vs a plugin, as it's not as requested as other stuff, and kind of reinforces the ["pets" mentality vs "cattle](https://blog.engineyard.com/2014/pets-vs-cattle) about hosts management . In any case:
- Making the host object implement this behavior seems wrong, I think it's the perfect example of a "God object". Facets, concerns, or even a service can do the job too. The "lock" feature doesn't need to know anything about a Host information. It's just an object that shouldn't be deleted. FWIW, it could be a string.
- We have similar behavior for provisioning templates, so I'd expect Hosts to follow a similar pattern with regards to the UI and the backend. So if this is happening please move that code to somewhere that both parts (Hosts/Templates) can use it. Rails provides part of this functionality through ActiveRecord `readonly?` method, check that out.

Let me know if that makes sense,

**Pull request closed without merging**